examples/source_examples/veeva_vault_objects/connector.py

```diff
@@ -0,0 +1,171 @@
+# This is a simple example for how to work with the fivetran_connector_sdk module.
+# This code is currently configured to Retrieve Details from All Object Types in Veeva Vault and then utilize VQL to retrieve all Object records, creating 1 table per object
+# You will need to provide your own Veeva Vault credentials for this to work --> vaultDNS, username, and password variables in configuration.json 
+# Retrieve Details from All Object Types endpoint: https://developer.veevavault.com/api/24.2/#retrieve-details-from-all-object-types
+# VQL endpoint: https://developer.veevavault.com/api/24.2/#vault-query-language-vql
+# Can also add code to extract from other endpoints as needed
+# See the Technical Reference documentation (https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update)
+# and the Best Practices documentation (https://fivetran.com/docs/connectors/connector-sdk/best-practices) for details
+
+from fivetran_connector_sdk import Connector
+from fivetran_connector_sdk import Operations as op
+from fivetran_connector_sdk import Logging as log
+import requests
+import json
+import datetime
+
+# Function to start a Veeva Session, inputs needed are: username, password, and vaultDNS -- all from the configuration.json file
+# Auth endpoint: https://developer.veevavault.com/api/24.2/#user-name-and-password 
+# The startVeevaSession function takes 1 parameter:
+# - configuration: a dictionary that holds the configuration settings for the connector.
+def startVeevaSession(configuration: dict):
+    username = configuration.get('username')
+    password = configuration.get('password')
+    auth_url = f"https://{configuration.get('vaultDNS')}/api/v24.2/auth"
+    headers = {
+        "Content-Type": "application/x-www-form-urlencoded",
+        "Accept": "application/json"
+    }
+    data = {
+        "username": username,
+        "password": password
+    }
+    response = requests.post(auth_url, headers=headers, data=data)
+    if response.status_code == 200:
+        response_json = response.json()
+        if response_json.get('responseStatus') == "SUCCESS":
+            session_id = response_json.get('sessionId')
+        else:
+            log.severe(f"Failed to create session for user with username: {username}")
+            session_id = None
+    else:
+        log.severe(f"Failed to get 200 response from Auth URL: {auth_url}, received response status code of {response.status_code}")
```

Comment:
polish this log statement, and also throw the response.message, so that error is visible

Reply:
Would this work? "Authentication failed: {auth_url} returned {response.status_code}. Expected 200."